src/modules/map/components/assistChips/SharePositionChip.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * @module modules/map/components/assistChips/SharePositionChip
+ */
+import { html } from '../../../../../node_modules/lit-html/lit-html';
+import { QueryParameters } from '../../../../domain/queryParameters';
+import { $injector } from '../../../../injection/index';
+import { openModal } from '../../../../store/modal/modal.action';
+import { isCoordinate } from '../../../../utils/checks';
+import { AbstractAssistChip } from '../../../chips/components/assistChips/AbstractAssistChip';
+import shareIcon from './assets/share.svg';
+
+const Update = 'update';
+
+/**
+ * A chip to share a specified position with a link. The link refers
+ * to the webapp itself with highlight-feature on the specified position.
+ * @class
+ * @property {module:domain/coordinateTypeDef~Coordinate} center The center coordinate of the shared position
+ * @author thiloSchlemmer
+ */
+export class SharePositionChip extends AbstractAssistChip {
+	constructor() {
+		super({ center: null });
+		const {
+			TranslationService: translationService,
+			EnvironmentService: environmentService,
+			UrlService: urlService,
+			ShareService: shareService
+		} = $injector.inject('TranslationService', 'EnvironmentService', 'UrlService', 'ShareService');
+		this._translationService = translationService;
+		this._environmentService = environmentService;
+		this._shareService = shareService;
+		this._urlService = urlService;
+	}
+
+	update(type, data, model) {
+		switch (type) {
+			case Update:
+				return {
+					...model,
+					center: data
+				};
+		}
+	}
+
+	getIcon() {
+		return shareIcon;
+	}
+
+	getLabel() {
+		const translate = (key) => this._translationService.translate(key);
+		return translate('map_assistChips_share_position_label');
+	}
+
+	isVisible() {
+		const { center } = this.getModel();
+		return isCoordinate(center);
+	}
+
+	async onClick() {
+		const { center } = this.getModel();
+		const useShareApi = this._environmentService.getWindow().navigator.share ? true : false;
+
+		const url = await this._buildShareUrl(center);
+		const shareAction = useShareApi ? (url) => this._shareUrlWithApi(url) : (url) => this._shareUrlDialog(url);
+		shareAction(url);
+	}
+
+	async _buildShareUrl(center) {
+		const extraParams = { [QueryParameters.CROSSHAIR]: true };
+		const url = new URL(this._shareService.encodeStateForPosition({ center: center }, extraParams));
+		try {
+			const shortUrl = await this._urlService.shorten(url.toString());
+			return shortUrl;
+		} catch (error) {
+			console.warn('Could not shorten url', error);
+			return url;
+		}
+	}
+
+	async _shareUrlWithApi(url) {
+		const translate = (key) => this._translationService.translate(key);
+
+		try {
+			await this._environmentService.getWindow().navigator.share({
+				title: translate('map_assistChips_share_position_link_title'),
+				url: url
+			});
+		} catch (error) {
+			console.error('Share-API failed:', error);
```

Comment:
Wonder if we should show a notification for that case

And also here:
[} catch (error) {](https://github.com/ldbv-by/bav4-nomigration/blob/ed18df85dd453bb1d064b650fb9151286358f9d4/src/modules/share/components/dialog/ShareDialogContent.js#L116-L118)